cmd/README.dynamodb.md

```diff
@@ -0,0 +1,141 @@
+would be much more expensive.
+
+Columns with consistent types are assigned Spanner types as detailed below.
+Columns without a consistent type are mapped to STRING.
+
+#### `Number`
+
+In most cases, we map the Number type in DynamoDB to Spanner's Numeric type.
+However, since the [range of Numeric](https://cloud.google.com/spanner/docs/storing-numeric-data)
+in Cloud Spanner is smaller than the [range of Number](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html)
+in DynamoDB, this conversion could result in out of range with potential
+precision loss. To address this possibility, we try to convert the sample data,
+and if it consistently fails, we choose STRING type for the column.
+
+#### `Null` Data Type
+
+In DynamoDB, a column can have a Null data type that represents an unknown or
+undefined state. Also, each row defines its own schema for columns (not for
+primary keys). So columns can be absent in rows. 
+
+We treat the above two cases the same as a Null value in Cloud Spanner. The
+cases that a column contains a Null value or a column is not present is an
+indication that this column should be nullable.
+
+#### `List` and `Map`
+
+In Cloud Spanner, the most similar type to List and Map is
+[STRUCT](https://cloud.google.com/spanner/docs/data-types#struct_type), but it
+is not a valid column type (available for query but not for storage).
+Therefore, we encode them into a json string. 
+
+#### Occasional Errors
+
+To prevent a few spurious rows from impacting schema construction, we define an
+error threshold: when building a type for a column, if the percentage of rows
+with a specific data type is lower than or equal to an extremely low
+value (0.1%), then we treat those rows as likely errors. Such rows are ignored
+for schema construction: their type is not considered a candidate type for the
+column.
+
+#### Multi-type Columns
+
+For a special scenario, we may get a column that has equal distribution of two
+data types. E.g., a column has 40% rows in String and 60% rows in Number. If we
+choose Number as its type, it may lead to 40% data loss in the data conversion.
+In the migration, we define a conflicting threshold on rows (after removing Null
+data types and rows that the column is not present). By default, the conflicting
+threshold is 5% and if the percentages of two or more data types are greater
+than it, we would consider that the column has conflicting data types. As a safe
+choice, we define this column as a STRING type in Cloud Spanner. 
+
+## Data Conversion
+
+### A Scan for Entire Table
+
+Data conversion proceeds table by table. For each table, we use the Scan API to
+read data. Each read has a size limit up to 1MB. By using the returned token, we
+make a subsequent call to continue retrieving data from the table.
+
+The row result contains the data type and data itself. According to our
+[inferred schema](#schema-inference), we will parse the row to a format that
```

Comment:
Broken link here as well. I guess all links that are relative links (starting with #) would fail

Reply:
These copies in cmd/ are only supposed to be used as a staging area until we are ready to copy them over to their original locations. Thus, it's okay if the links are temporary broken. We will get an opportunity to review the same again in a separate PR focused on updating READMEs. I will add a temporary banner to top of these READMEs to reflect the same.